BZ-21767: feat: Add-support-for-animation-direction-in-toast-component

badri-singhal opened the pull request:
- Directions supported is left to right, right to left, top to bottom, bottom to top
- Toast position can be over content or inside content
- Dev Proof:- https://drive.google.com/file/d/1KjDnA8ywAX2kK2ZbvOq60xxgMMuv5MnN/view?usp=sharing

AmreetKumarkhuntia commented on src/lib/Toast/Toast.svelte:
```diff
@@ -6,6 +6,22 @@
+    'bottom-to-top': { in: { x: 0, y: 500, duration: 400 }, out: { x: 0, y: 500, duration: 800 } }
+  };
+
+  const inAnimationConfig = animationConfig[properties.direction || 'top-to-bottom'].in;
```
nit : can pass default and use variable

swaroopvarma1 commented on src/lib/Toast/Toast.svelte:
```diff
@@ -1,11 +1,44 @@
+    }
+  }
+
+  const inAnimationConfig = getAnimationConfig(properties.direction).in;
```
no need of multiple variable initialization

swaroopvarma1 commented on src/lib/Toast/Toast.svelte:
```diff
@@ -1,11 +1,43 @@
   export let properties: ToastProperties = defaultToastProperties;
   const dispatch = createEventDispatcher();
 
+  function getAnimationConfig(toastDirection: ToastDirection | undefined) {
```
null

swaroopvarma1 commented on src/lib/Toast/Toast.svelte:
```diff
@@ -1,11 +1,48 @@
   export let properties: ToastProperties = defaultToastProperties;
   const dispatch = createEventDispatcher();
 
+  interface AnimationConfig {
```
type

sinha-sahil commented on src/lib/types.ts:
```diff
@@ -36,3 +36,11 @@ export type TextTransformer = (text: string) => string;
+ * @description Type for animation configuration
+ */
+export type FlyAnimationConfig = {
+  in: { x: number; y: number; duration: number };
```
This `{ x: number; y: number; duration: number }` can be a type and can be used with `in` & `out`.
Lets check if svelte transition exposes this type, if yes, lets use it.

sinha-sahil commented on src/lib/Toast/properties.ts:
```diff
@@ -1,11 +1,18 @@
   rightIcon?: string;
   type?: ToastType;
+  direction?: ToastDirection;
+  overContent?: boolean;
```
`overlapPage` sounds more expressive.

sinha-sahil commented on src/lib/Toast/Toast.svelte:
```diff
@@ -66,11 +115,20 @@
     background-color: var(--default-background-color, #87ceeb);
   }
+
+  .overContent {
```
classes must be in `kebab-case`

sinha-sahil commented on src/lib/Toast/Toast.svelte:
```diff
@@ -31,9 +78,11 @@
-    in:fly={{ x: 500, y: 0, duration: 400 }}
-    out:fly={{ x: 500, y: 0, duration: 800 }}
+    class="toast {properties.type ?? ''} {properties?.overContent === false
+      ? 'insideContent'
```
lets add separate class check for neatness.

`class:overlap-content={...check_for_should_overlap...}`

sinha-sahil commented on src/lib/Toast/Toast.svelte:
```diff
@@ -17,6 +22,48 @@
+    let outX: number = 0;
+    let outY: number = 0;
+
+    switch (toastDirection) {
```
Lets add comments on whats being done here.

sinha-sahil commented on src/lib/Toast/Toast.svelte:
```diff
@@ -32,8 +93,9 @@
     class="toast {properties.type ?? ''}"
-    in:fly={{ x: 500, y: 0, duration: 400 }}
-    out:fly={{ x: 500, y: 0, duration: 800 }}
+    class:inside-content={!(properties?.overlapPage ?? true)}
```
`no-page-overlap`

badri-singhal committed:
BZ-21767: feat: Added customization support for toast component

- Added support for customizing direction for toast
  - entry direction
  - entry animation
  - positioning
